src/components/Calculator.js

```diff
@@ -1,30 +1,50 @@
 import React from 'react';
+import calculate from '../logic/calculate';
 
 export default class Calculator extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+    };
+  }
+
+  clickListener = (e) => {
+    e.preventDefault();
+    const result = e.target.value;
+    const obj = calculate(this.state, result);
+    this.setState(obj);
+  };
+
   render() {
```

Comment:
Great design @businesskaniki 🤝

- [ ] [Optional] Kindly use ES6 syntax for all your functions and methods.